workflows/phylorthology.nf

```diff
@@ -48,66 +70,416 @@ include { INPUT_CHECK } from '../subworkflows/local/input_check'
+    ch_dmd = ORTHOFINDER_PREP.out.dmd.flatten()
+    ch_test_fa = ORTHOFINDER_PREP_TEST.out.fa.flatten()
+    ch_test_dmd = ORTHOFINDER_PREP_TEST.out.dmd.flatten()
 
     //
     // SUBWORKFLOW: Read in samplesheet, validate and stage input files
     //
-    INPUT_CHECK (
+    // First the full set
+    ch_prots_all = INPUT_CHECK (
         ch_input
     )
+    .prots
+    .map {
+        meta, prots ->
+            def meta_clone = meta.clone()
+            meta_clone.id = meta_clone.id.split('_')[0..-2].join('_')
+            [ meta_clone, prots ]
+    }
+    .groupTuple(by: [0])
+
+    // Pull out the test and full sets
+    ch_prots_all
+    .branch {
+        meta, prots ->
+            proteomes  : meta.mcl_test == 'false'
+                return [ meta, prots.flatten() ]
+    }
+    .set { ch_prots_full }
+    
+    ch_prots_all
+    .branch {
+        meta, prots ->
+            proteomes  : meta.mcl_test == 'true'
+                return [ meta, prots.flatten() ]
+    }
+    .set { ch_prots_test }
+    
+    ch_prots_full
+    .branch {
+        meta, prots ->
+            proteomes  : prots
+                return [ meta, prots.flatten() ]
+    }
+    .set { ch_busco_full }
+
+    ch_prots_test
+    .branch {
+        meta, prots ->
+            proteomes  : prots
+                return [ meta, prots.flatten() ]
+    }
+    .set { ch_busco_test }
+    
+    // Create an orthofinder channel with paths to the new fasta/diamond DBs
+    ch_orthofinder_full = ch_busco_full.merge(ch_fa, ch_dmd)
+    ch_orthofinder_test = ch_busco_test.merge(ch_test_fa, ch_test_dmd)
     ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
 
+    // Now, there will be a couple of modules below that we reapply, both to 
+    // the full dataset, and to the MCL inflation parameter test set. 
+    // These repeat modules include:
+    // uniprot annotation (ch_annotations)
+    // diamond blastp (ch_blastp & ch_similarities)
+    
+    //
+    // MODULE: Annotate UniProt Proteins
+    //
+    ch_annotations = ANNOTATE_UNIPROT (
+            ch_busco_full
+        )
+        .annotations
+        .collect()
+    ch_versions = ch_versions.mix(ANNOTATE_UNIPROT.out.versions)
+    
+    //
+    // MODULE: Run BUSCO
+    // Split up into shallow and broad scale runs, since downstream modules
+    // do not use these outputs, so multiple busco runs may be conducted
+    // simultaneously
+    //
+    // Shallow taxonomic scale:
+    BUSCO_SHALLOW (
+        ch_busco_full,
+        "shallow",
+        [],
+        []
+        )
+    
+    // Broad taxonomic scale (Eukaryotes)
+    BUSCO_BROAD (
+        ch_busco_full,
+        "broad",
+        [],
+        []
+        )
+        
     //
-    // MODULE: Run FastQC
+    // MODULE: All-v-All diamond/blastp
     //
-    FASTQC (
-        INPUT_CHECK.out.reads
+    // Run for the test set (used to determine the best value of the MCL
+    // inflation parameter)
+    ch_blastp_test = DIAMOND_BLASTP_TEST (
```

Comment:
I can use different language. I called it test for brevity, but the "test" dataset here is the one used for MCL inflation parameter testing. Because I'm running the mcl clustering twice, that's why I needed the two datasets to be analyzed within the same workflow. 

Basically what I was anticipating is that the user would specify a subset of species (ideally for which we have uniprot protein accessions) and use that subset of samples for mcl testing. That subset would be indicated in the samplesheet. 

Currently I have the test samples stored in a separate directory - the samplesheet lists the samples twice, one in the directory where the complete dataset is stored, and one where the MCL testing set is stored. A column indicates whether each sample is to be used for MCL testing or not.

Reply:
Ok so in the "test" dataset nothing is physically different from the fasta files that are in this test dataset and the fasta files in the full dataset - there are just additional species fasta files in the full dataset? 

nf-core convention is that a "test" dataset is literally a pared down dummy dataset of real files - like the example above. Just to ensure the workflow works as intended and completes all the processes to the end. 

Here I think your test and full are really representative species and then the MCL scores are propagated to other species for the full run. I think there is similar nomenclature used in this PyParanoid pipeline: https://github.com/ryanmelnyk/PyParanoid where test = training dataset, and you propagate the scores to the entire dataset or additional genomes down the line

Reply:
Correct! It's just a pared down number of samples, but it's a "real" dataset to be used to inform downstream inflation parameter selection. 

I'm happy to rename it - I worry a bit by calling it a "training" dataset, since that language suggests there's some sort of machine learning analysis going on here, but that might be an unnecessary worry.